Beds 1350/cl deposits table pectra

enzo-bitfly opened the pull request:
Based on #1414

enzo-bitfly committed:
feat: add `BcIconFlag`

enzo-bitfly commented on frontend/components/dashboard/table/DashboardTableClDeposits.vue:
```diff
@@ -130,221 +109,257 @@ const {
+            </template>
+          </Column>
+          <template #expansion="slotProps">
+            <div class="cl-deposits-table__details">
+              <div class="cl-deposits-table__details-grid">
```
thought: I'm considering abstracting the expandable row into a component since all new tables have the same style.

marcel-bitfly commented on frontend/components/bc/icon/BcIconFlag.vue:
**praise:** thx for minifying via https://svgomg.net/

marcel-bitfly commented on frontend/components/bc/icon/BcIcon.vue:
```diff
@@ -132,6 +133,7 @@ const is = computed(() => {
     <LazyBcIconEye v-else-if="name === 'eye'" />
     <LazyBcIconFileSignature v-else-if="name === 'file-signature'" />
     <LazyBcIconFire v-else-if="name === 'fire'" />
+    <LazyBcIconFlag v-else-if="name === 'flag'" />
```
**thought (non-blocking):** noooo my naming system is crashed 😱

<BcIconFlag> should be the parent of <BcIconFlagUs> and so on...we might have to have another look at this at another time

marcel-bitfly commented on frontend/composables/useDashboardData.ts:
**thought (non-blocking):** since this is not a `composable` we should remove the `use`-prefix

you could probably add it to `/utils`

marcel-bitfly commented on frontend/pages/dashboard/[[id]]/index.vue:
```diff
@@ -207,7 +207,6 @@ const dashboardData = useDashboardData()
 const elDepositsQueryParams = ref<TableQueryParams>({
   limit: 5, sort: 'timestamp:desc',
 })
-
```
**nitpick:** plz try to reduce line noise (except managing whitespace deliberately)

enzo-bitfly replied:
sorry about this one. I need it for my inner monk

marcel-bitfly commented on frontend/pages/dashboard/[[id]]/index.vue:
```diff
@@ -285,9 +315,9 @@ const refreshActiveTab = () => {
-      </div>
+
+      <!-- <DashboardSlotViz /> -->
+
```
**issue:** 🫣

enzo-bitfly replied:
Done

marcel-bitfly commented on frontend/components/dashboard/table/DashboardTableClDeposits.vue:
```diff
@@ -1,97 +1,72 @@
-    signature: width.value >= 1100,
-    slot: width.value >= 900,
-    withdrawalCredentials: width.value >= 800,
+    group: width.value > 768,
+    publicKey: width.value >= 768,
+    status: width.value >= 768,
+    type: width.value >= 768,
+    validatorIndex: width.value >= 768,
+    withdrawalCredentials: width.value >= 768,
```
**question:** did you check with @philip-bitfly that this is correct?

enzo-bitfly replied:
Yes, we discussed this

enzo-bitfly committed:
feat: adapt `CL deposits table` for Pectra

-Move fetching logic to `dashboard` root page component
-Ensure immediate fetch for active tab on first page load

See: BEDS-1362
